vividus-engine/src/main/java/org/vividus/replacement/DeprecatedStepNotificationFactory.java

```diff
@@ -45,8 +46,8 @@ public DeprecatedStepNotificationFactory(RunTestContext runTestContext, Configur
     protected String createDeprecatedStepNotification(String versionToRemove, String replacementFormatPattern)
     {
         String runningDeprecatedStep = runTestContext.getRunningStory().getRunningSteps().getFirst();
-        Matcher stepMatcher = getStepMatcher(runningDeprecatedStep);
-        String[] stepParams = getParametersFromStep(stepMatcher);
+        Optional<Matcher> stepMatcher = getStepMatcher(runningDeprecatedStep);
+        String[] stepParams = stepMatcher.map(this::getParametersFromStep).orElseGet(() -> new String[0]);
```

Comment:
that's a fix of the reason, but not root cause, any plans to do root cause analysis?

Reply:
> that's a fix of the reason, but not root cause, any plans to do root cause analysis?

The root of the problem is that steps within composite steps can be processed by separate logic (without using `stepMonitor` which we use to collect patterns of the current steps from tests) on the JBehave side.
https://github.com/jbehave/jbehave-core/blob/9510692f56f31551aa52e0eeddc03cc426deaea5/jbehave-core/src/main/java/org/jbehave/core/steps/StepCandidate.java#L242-L243

I'll think about how to make the "monitoring" of matched steps more unified on the JBehave side.